src/main/java/com/github/princesslana/eriscasper/gateway/Gateway.java

```diff
@@ -85,14 +88,31 @@ private void sequenceNumberSeen(Optional<SequenceNumber> seq) {
   }
 
   @SuppressWarnings("unchecked")
-  public Observable<Event> connect(String url, BotToken token) {
+  public Observable<Event> connect(String url, BotToken token, Optional<Shard> shard) {
     Observable<Payload> ps =
         ws.connect(String.format("%s?v=%s&encoding=%s", url, VERSION, ENCODING))
             .ofType(RxWebSocketEvent.StringMessage.class)
             .map(RxWebSocketEvent.StringMessage::getText)
             .flatMapMaybe(
                 Singles.toMaybeAnd(
                     payloads::read, (s, t) -> LOG.warn("Error reading payload: {}", s, t)))
+            .takeUntil(
+                payload -> {
+                  if (payload.op().getCode() == 9) {
```

Comment:
Do we not have an enum or constants around instead of having to use 9 here?

Reply:
I'll change it